src/tasks/bundler.rs

```diff
@@ -1,58 +1,86 @@
 //! Bundler service responsible for managing bundles.
+use std::sync::Arc;
+
 use super::oauth::Authenticator;
 
 pub use crate::config::BuilderConfig;
 
+use alloy::consensus::TxEnvelope;
+use alloy::eips::eip2718::Encodable2718;
+use alloy::rpc::types::mev::EthSendBundle;
 use oauth2::TokenResponse;
 use reqwest::Url;
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
-use std::time::{Duration, Instant};
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver};
+use tokio::task::JoinHandle;
 use zenith_types::ZenithEthBundle;
 
-/// A bundle response from the tx-pool endpoint, containing a UUID and a
-/// [`ZenithEthBundle`].
+/// Holds a Signet bundle from the cache that has a unique identifier
+/// and a Zenith bundle
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Bundle {
-    /// The bundle id (a UUID)
+    /// Cache identifier for the bundle
     pub id: String,
-    /// The bundle itself
+    /// The Zenith bundle for this bundle
     pub bundle: ZenithEthBundle,
 }
 
+impl PartialEq for Bundle {
+    fn eq(&self, other: &Self) -> bool {
+        self.id == other.id
+    }
+}
+
+impl Eq for Bundle {}
+
+impl From<TxEnvelope> for Bundle {
+    fn from(tx: TxEnvelope) -> Self {
+        let tx_vec = vec![tx.encoded_2718().into()];
+        Self {
+            id: tx.tx_hash().to_string(),
```

Comment:
cc @Evalir the hash is not the cache id, right? 

Unless i'm mistaken, there is no local conversion possible from a transaction to a bundle, and this impl should be deleted